packages/core/test/integration/dialects/mssql/query-error.test.ts

```diff
@@ -0,0 +1,162 @@
+import { assert, expect } from 'chai';
+import {
+  AggregateError,
+  DataTypes,
+  ForeignKeyConstraintError,
+  UnknownConstraintError,
+  ValidationError,
+} from '@sequelize/core';
+import { sequelize } from '../../support';
+
+const queryInterface = sequelize.queryInterface;
+const dialect = sequelize.getDialect();
+
+describe('[MSSQL Specific] Query Errors', () => {
```

Comment:
Do you happen to know how much these expectations change per dialect? Maybe it's nice to test these on all dialects to see how our implementation changes

Reply:
I've merged the tests with the existing error test file.
LMK what you think.